index.bs

```diff
@@ -5667,6 +5678,303 @@ The [=remote end subscribe steps=], with [=subscribe priority=] 10, given
 
 </div>
 
+## Input ## {#module-input}
+
+The <dfn export for=modules>input</dfn> module contains functionality for
+simulated user input.
+
+### Definition ### {#module-input-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+InputCommand = (
+  input.PerformActionsCommand //
+  input.ReleaseActionsCommand
+)
+</pre>
+
+### Types ### {#module-input-types}
+
+#### input.ElementOrigin ### {#module-input-origin}
+
+Tjhe <code>input.ElementOrigin</code> type represents an {{Element}} that will
+be used as a coordinate origin.
+
+<pre class="cddl remote-cddl">
+input.ElementOrigin = {
+  "type": "node",
```

Comment:
We check that the value matches this specific production, which it can't if the `type` is missing. So I don't quite understand the concern?

Reply:
For `script.callFunction` we can pass-in any object that only needs `sharedId` to have set and we deserialize it as a `Node`. So the type there (`SharedReference`) is optional while here it is required.

Reply:
Oh, and this probably needs to be `Extensible`? A client should not have to modify a node reference that it got back from eg `script.evaluate` before it can be used for an action.

Reply:
I want `type` here so that in the future we have the option to extend to other origin types, without having to inspect more than the type field.

I'd happily make the same change for `SharedReference`.

Reply:
But isn't this `type` a problem when we deserialize a `SharedReference` then? Right now it's useless. Wouldn't it make more sense instead to make this change together with the SharedReference one in a follow-up PR?

Reply:
If we just have a `sharedId` field and we want to add another kind of element-offset origin in the future (e.g. one related to the top left of the element rather than the centre point), what happens? We can't just add a `type` field because that will be ignored by existing clients (assuming this is `Extensible`). So in that case we'd have to use a different field name to represent the `sharedId`. But of course that would break the property that you can just directly reuse an element that you got as a response as the origin. So I think fundamentally that naming variants gives us a much more extensible protocol, at the cost of clients needing some explicit typecasts.

Reply:
Thanks for the update James! With the wrapping this looks fine to me now. 

@sadym-chromium maybe you can have a look at the recent changes as well?